src/field/pokemon.ts

```diff
@@ -3648,9 +3717,12 @@ export abstract class Pokemon extends Phaser.GameObjects.Container {
    * @see {@linkcode MoveRestrictionBattlerTag}
    */
   isMoveTargetRestricted(moveId: MoveId, user: Pokemon, target: Pokemon): boolean {
-    for (const tag of this.findTags((t) => t instanceof MoveRestrictionBattlerTag)) {
-      if ((tag as MoveRestrictionBattlerTag).isMoveTargetRestricted(moveId, user, target)) {
-        return (tag as MoveRestrictionBattlerTag) !== null;
+    const restrictingTags = this.findTags<MoveRestrictionBattlerTag>((t) =>
+      t.isType<MoveRestrictionBattlerTag>(...RESTRICTING_TAG_TYPES),
+    );
+    for (const tag of restrictingTags) {
+      if (tag.isMoveTargetRestricted(moveId, user, target)) {
+        return tag !== null; // TODO: this doesn't seem correct, `Pokemon#findTags` can't return `null`
       }
     }
     return false;
```

Comment:
```suggestion
    return restrictingTags.some((tag) => tag.isMoveTargetRestricted(moveId, user, target));
```
`tag !== null` should always be `true`, so this is equivalent.